domain/src/Commands/DomainCommands.php

```diff
@@ -0,0 +1,1288 @@
+    // Try to avoid repeated calls to getFieldMap() assuming it's expensive.
+    $this->ensureEntityFieldMap();
+    return array_key_exists($field, $this->entity_field_map[$entity_type]);
+  }
+
+  /**
+   * Ensure the local entity field map has been defined.
+   *
+   * @return string[]
+   *   List of entities supporting domain access.
+   */
+  protected function ensureEntityFieldMap() {
+    // Try to avoid repeated calls to getFieldMap() assuming it's expensive.
+    if (empty($this->entity_field_map)) {
+      $entity_manager         = \Drupal::entityManager();
+      $this->entity_field_map = $entity_manager->getFieldMap();
+    }
+  }
+
+  /**
+   * Enumerate entities of the supplied type and domain.
+   *
+   * @param string $entity_type
+   *   The entity type name, e.g. 'node'
+   * @param string $domain
+   *   The machine name of the domain to enumerate.
+   * @param string $field
+   *   The field to manipulate in the entity, e.g. DOMAIN_ACCESS_FIELD.
+   *
+   * @return int|string[]
+   *   List of entity IDs for the selected domain.
+   * @todo: should this really be a string[] of fields?
+   */
+  protected function enumerateDomainEntities(string $entity_type, string $domain, string $field, $just_count = FALSE) {
+    if (!$this->entityHasDomainField($entity_type, $field)) {
+      $this->logger()->info("Entity type @entity_type does not have field @field, so none found.",
+        ['@entity_type'=> $entity_type,
+         '@field' => $field]);
+      return [];
+    }
+
+    $efq = \Drupal::entityQuery($entity_type);
+    // Dont access check or we wont get all of the possible entities moved.
+    $efq->accessCheck(FALSE);
+    $efq->condition($field, $domain, '=');
+    if ($just_count) {
+      $efq->count();
+    }
+    return $efq->execute();
+  }
+
+  /**
+   * Reassign old_domain entities, of the supplied type, to the new_domain.
+   *
+   * @param string $entity_type
+   *   The entity type name, e.g. 'node'
+   * @param string $field
+   *   The field to manipulate in the entity, e.g. DOMAIN_ACCESS_FIELD.
+   *   @todo: should this really be a string[] of fields?
+   * @param \Drupal\domain\DomainInterface $old_domain
+   *   The domain the entities currently belong to. It is not an error for
+   *   entity ids to be passed in that are not in this domain, though of course
+   *   not very useful.
+   * @param \Drupal\domain\DomainInterface $new_domain
+   *   The domain the entities should now belong to: When an entity belongs to
+   *   the old_domain, this domain replaces it.
+   * @param array $ids
+   *   List of entity IDs for the selected domain and all of type $entity_type.
+   *
+   * @return int
+   *
+   * @throws \Drupal\Component\Plugin\Exception\PluginException
+   * @throws \Drupal\Core\Entity\EntityStorageException
+   */
+  protected function reassignEntities(string $entity_type, string $field, DomainInterface $old_domain, DomainInterface $new_domain, $ids) {
```

Comment:
Type hint `array $ids`

Reply:
This method also might move to the DomainElementManager, which needs a slight docs update to say `Handles field management for domain entity references.`